library/src/window_impl.cxx

```diff
@@ -44,6 +55,97 @@ class window_impl::internals
     return this->CachePath;
   }
 
+#if _WIN32
+  /**
+   * Helper function to detect if the
+   * Windows Build Number is equal or greater to a number
+   */
+  BOOL IsWindowsBuildNumberOrGreater(int buildNumber)
+  {
+    int windowsBuildNumber = 0;
+
+    // Receives the handle to Reg key
+    HKEY versionHKey;
+    LONG result = RegOpenKeyExA(HKEY_LOCAL_MACHINE,
+      "SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion", 0, KEY_READ, &versionHKey);
+
+    // Query value from the reg key
+    if (result == ERROR_SUCCESS)
+    {
+      // Receives code indicating the type of data stored in the reg value.
+      // CurrentBuildNumber is of type REG_SZ
+      DWORD type;
+      DWORD dataSize = MAX_VALUE_NAME;
+      CHAR data[MAX_VALUE_NAME]{};
```

Comment:
is that a 16kB buffer to hold a text representation of an `int`?

Reply:
So, I looked up registry value and `BuildNumber` is a string type. `REG_SZ`. So I am creating a buffer for it.

This constant is taken from a windows example. 

Found an article on windows registry using modern c++. https://learn.microsoft.com/en-us/archive/msdn-magazine/2017/may/c-use-modern-c-to-access-the-windows-registry

Reply:
I am now using vtksys wrapper to handle string values